File: Prototype3/Frontend/tapatapp_flutter/lib/core/models/comment.dart
import 'package:json_annotation/json_annotation.dart';

part 'comment.g.dart';

@JsonSerializable()
class Comment {
  final int id;
  @JsonKey(name: 'child_id')
  final int childId;
  @JsonKey(name: 'user_id')
  final int userId;
  final String text;
  final String timestamp;
  final bool important;

  Comment({
    required this.id,
    required this.childId,
    required this.userId,
    required this.text,
    required this.timestamp,
    this.important = false,
  });

  factory Comment.fromJson(Map<String, dynamic> json) => _$CommentFromJson(json);
  Map<String, dynamic> toJson() => _$CommentToJson(this);
}

File: ProtypePruebas/Backend/Backend/models/comment.py
from datetime import datetime

class Comment:
    def __init__(self, id, child_id, user_id, text, important, timestamp=None):
        self.id = id
        self.child_id = child_id
        self.user_id = user_id
        self.text = text
        self.important = bool(important)
        self.timestamp = timestamp if timestamp else datetime.now()

    def to_dict(self):
        return {
            "id": self.id,
            "child_id": self.child_id,
            "user_id": self.user_id,
            "text": self.text,
            "important": self.important,
            "timestamp": self.timestamp.isoformat() if self.timestamp else None
        }